_posts/2017-02-29-upgrade-to-babel-7.md

```diff
@@ -0,0 +1,228 @@
+
+This is when you are using `babel-plugin-transform-object-rest-spread`
+
+Before:
+
+```js
+var { ...y, } = { a: 1};
+```
+
+This will now throw a SyntaxError.
+
+After:
+
+```js
+var { ...y } = { a: 1};
+```
+
+or:
+
+```js
+var { ...y, b } = { a: 1};
+```
+
+## babel-preset-stage-1/babel-preset-stage-2 (decorators)
+
+> [legacy-decorators](https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy) has been moved into the [transform-decorators](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-decorators) package [#5225](https://github.com/babel/babel/pull/5225) ![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+We don't currently have a Stage 2 transform for decorators so instead of making it error, we are adding what was legacy-decorators as part of the Stage 1 preset by merging it into the transform-decorators plugin.
+
+## babel-core
+
+> `babel-core/register.js` has been removed [#5132](https://github.com/babel/babel/pull/5132) ![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+The deprecated usage of `babel-core/register` has been removed in Babel 7; instead use the standalone package `babel-register`.
+
+Install `babel-register` as a new dependency:
+
+```sh
+npm install --save-dev babel-register
+```
+
+Upgrading with Mocha:
+
+```sh
+mocha --compilers js:babel-core/register
+```
+
+to:
+
+```sh
+mocha --compilers js:babel-register
+```
+
+See [babel-register documentation](https://babeljs.io/docs/usage/babel-register/) for more information.
+
+## babel-plugin-transform-class-constructor-call
+
+> babel-plugin-transform-class-constructor-call has been removed [#5119](https://github.com/babel/babel/pull/5119) ![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+TC39 decided to drop this proposal.
+
+Before:
+
+```js
+class Point {
+
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+
+  call constructor(x, y) {
+    return new Point(x, y);
+  }
+
+}
+
+let p1 = new Point(1, 2);
+let p2 = Point(3, 4);
+```
+
+You can move your logic into the constructor or into a static method.
+
+Example with a static method:
+
+```js
+class Point {
+
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+
+  static secondConstructor(x, y) {
+    return new Point(x, y);
+  }
+
+}
+
+let p1 = new Point(1, 2);
+let p2 = Point.secondConstructor(3, 4);
+```
+
+See [/docs/plugins/transform-class-constructor-call/](/docs/plugins/transform-class-constructor-call/) for more information.
+
+## babel
+
+> Dropping the `babel` package [#5293](https://github.com/babel/babel/pull/5293) ![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+This package currently gives you an error message to install `babel-cli` instead in v6. We will just not publish a v7 version. It also doesn't make sense if we switch to scoped package `babel` -> `@babel/babel`?
```

Comment:
Use an emoji instead of ->? 😂 It might not work if the website currently doesn't support emojis.